src/main/java/com/uber/cadence/internal/worker/WorkflowWorker.java

```diff
@@ -287,21 +282,15 @@ public HistoryEvent next() {
           if (current.hasNext()) {
             return current.next();
           }
-          Duration passed = Duration.ofMillis(System.currentTimeMillis() - start);
-          Duration timeout = Duration.ofSeconds(startedEvent.getTaskStartToCloseTimeoutSeconds());
-          Duration expiration = timeout.minus(passed);
-          if (expiration.isZero() || expiration.isNegative()) {
-            throw new Error("History pagination time exceeded TaskStartToCloseTimeout");
-          }
 
           options.getMetricsScope().counter(MetricsType.WORKFLOW_GET_HISTORY_COUNTER).inc(1);
           Stopwatch sw =
               options.getMetricsScope().timer(MetricsType.WORKFLOW_GET_HISTORY_LATENCY).start();
           RetryOptions retryOptions =
               new RetryOptions.Builder()
-                  .setExpiration(expiration)
-                  .setInitialInterval(Duration.ofMillis(50))
-                  .setMaximumInterval(Duration.ofSeconds(1))
+                  .setExpiration(RetryServiceOperationExpirationInterval)
```

Comment:
Why did you remote the expiration calculation from the decision task timeout? Retrying for a minute when decision task has 10 second timeout is incorrect.

Reply:
changed it to use the timeout on the DecisionTaskScheduled Event